spec/schemas.spec.js

```diff
@@ -759,7 +759,7 @@ describe('schemas', () => {
     });
   });
 
-  it('refuses to put to existing fields, even if it would not be a change', done => {
+  it('refuses to put to existing fields with different type, even if it would not be a change', done => {
```

Comment:
@davimacedo raised the [question](https://github.com/parse-community/parse-server/pull/7091/files#r551768714) wether this could have side effects for dashboard - any we can think ok?

Reply:
Yes, most likely, i guess we could expose something (or an error message) to the dashboard to `lock` all schema modifications ?

When you restart the server,  changes made via the dashboard would be overriten by the migrations.

Reply:
Maybe we wouldn't even need a dashboard PR for this immediately, but couldn't we just block the create / update schema endpoints in parse server if schema is defined manually? That should at least display some error message, not only in dashboard, but also for any SDK that attempts to change that.

Reply:
1) 
Tested it and you can see the error in the console, there is probably a small thing missing to display the error in the little red status box on the dashboard

![image](https://user-images.githubusercontent.com/794449/129115053-6c824902-0708-42db-be43-c7b1c5bcda65.png)

2) 
I think we should disable `checkIfDefinedSchemasIsUsed()` for now for ~~two~~ three reasons.

a) Migrations will **not** delete fields, or collections when a schema is removed **unless* specified explicitly in the config. So you need to go in the dashboard to delete theses extra fields/collections.

b)  The migration will log in the console a nice message saying that some extra stuff is present. So i'm not sure we need this.
![image](https://user-images.githubusercontent.com/794449/129115665-880151ba-1517-4383-8cf2-c5ab876eeacf.png)


c) It was not working before my last commit, and i've start to use it for the last 15 minutes and it's REALLY ANOYING ! hehehe.

3)  
Another option would be to add a config option that is disabled by default `lockSchemas: false`, but IMO its not necessary, and we can always add this feature back if needed later.

Reply:
> why should it still be possible to do schema manipulations

I would say for any development related changes. The Dashboard is really helpful, and blocking **all** changes to the schema is really really annoying and frustrating.

> What other way does a developer have to see which fields and collections violate the defined schema?

Currently only console logging during startup. For now, i think it's suffisent, Later, we could add a mode to **dump current DB schema  as migrations**  a bit like [Doctrine migration generation](https://www.doctrine-project.org/projects/doctrine-migrations/en/1.7/reference/generating_migrations.html). Maybe adding a tab in the dashboard that will dump all deltas between declared schema migrations and current schemas, and allow user to copy/paste the updated schema. This will be really cool for users that dont have any schema declared and want to dump  the current existing DB schemas to put it in source control.


-----
In 9db55b63cf7635f87322efd0de7a72ae236dc3ef, i added the option `lockSchemas` that will turn the lock ON and prevent any outside changes. 

Once it get merged, we will also be able to update `parse-server-example` and add the option to demonstrate its use 

```js
serverConfig = {
   migrations: {
      lockSchemas: process.env.NODE_ENV === "production",
      schemas:  [ /* ....... */]
   }
}
```

Reply:
> Maybe adding a tab in the dashboard that will dump all deltas between declared schema migrations and current schemas, and allow user to copy/paste the updated schema. 

I like that!

> In 9db55b6, i added the option lockSchemas that will turn the lock ON and prevent any outside changes.

That's great, I think we can go forward like that.